okapi-test-header-module/src/test/java/HeaderModuleTest.java

```diff
@@ -78,11 +81,19 @@ public void test3(TestContext context, Async async) {
     HashMap<String, String> headers = new HashMap<>();
 
     OkapiClient cli = new OkapiClient(URL, vertx, headers);
-    cli.post("/_/tenantPermissions", "a, b,\nc", res -> {
-      cli.close();
-      context.assertTrue(res.succeeded());
-      context.assertEquals("a, b, c", cli.getRespHeaders().get("X-Tenant-Perms-Result"));
-      async.complete();
+    cli.post("/_/tenantPermissions", "{", res1 -> {
+      context.assertTrue(res1.failed());
+      JsonObject perm = new JsonObject("{\"k\": \"v\"}");
+      cli.post("/_/tenantPermissions", perm.encode(), res2 -> {
+        context.assertTrue(res2.succeeded());
+        context.assertEquals("GET test-header-module /_/tenantPermissions 200 -",
+            cli.getRespHeaders().get(XOkapiHeaders.TRACE));
+        cli.get("/permResult", res3 -> {
+          cli.close();
+          context.assertEquals(perm.encode(), new JsonArray(res3.result()).getJsonObject(0).encode());
```

Comment:
encode() does not guarantee the order of the keys. You probably want to compare without encode().

Reply:
yes.